### ISSUE TITLE
tf2pulumi should support tf 0.12 syntax

### ISSUE THREAD

**Issue by davidobrien1985** [pulumi/tf2pulumi#136]
Our customer has previously written a lot of their terraform scripts in 0.12 syntax.
We are now introducing pulumi to them and were planning to use `tf2pulumi` for the majority, however, this doesn't seem to support 0.12 syntax yet.

`error: creating tree module: Error parsing 03-main.tf: At 7:21: Unknown token: 7:21 IDENT var.ARM_SUBSCRIPTION_ID`

**Comment by pgavlin** [comment]
We're working on rudimentary support for this now.

If you can share the sort of TF12 constructs the code you'd like to convert uses, that would help me in prioritizing the specific work we need to do.

**Comment by davidobrien1985** [comment]
@pgavlin 
Generally speaking it's the following in my case:
* no more requirement to use `${}` to reference variables or other resources

Instead of `ARM_ENVIRONMENT_NAME          = ${var.ARM_ENVIRONMENT_NAME}` we can now do `ARM_ENVIRONMENT_NAME          = var.ARM_ENVIRONMENT_NAME`

That's the only one that we're mainly using right now. I don't think we're using others.

**Comment by jnevins-gcm** [comment]
also unsupported
```
variable "shares" {
  type = list(object({
    name  = string
    quota = number
  }))
}
```

**Comment by jnevins-gcm** [comment]
any update on this? would really help motivate me to convert

**Comment by pgavlin** [comment]
It's very much on our radar. I don't have a firm target date, but it's something we'd like to address as soon as possible.

The distance between TF11 and TF12--particularly in terms of the APIs used to represent the TF config--is large enough that supporting TF12 isn't as simple as I had originally hoped.

**Comment by swoop-alex** [comment]
is this going to be fixed with milestone 0.32  Due by March 05, 2020?

**Comment by pgavlin** [comment]
Though I'm actively working on this, it is unlikely to be finished by the end of 0.32. We have committed to delivering it by the end of 0.33.

**Comment by JeffN825** [comment]
Is there a way to use the work in progress?

Also - how does this handle tf modules?

**Comment by lukehoban** [comment]
Work in progress on this in:

- [x] https://github.com/pulumi/pulumi/pull/3980 
- [x] https://github.com/pulumi/pulumi/pull/3984
- [ ] https://github.com/pulumi/pulumi/pull/3999
- [ ] Integrate into `tf2pulumi` frontend

> Is there a way to use the work in progress?

Not yet, but we will update this issue once there is a working version of the TF0.12 support.

**Comment by domingoladron** [comment]
@davidobrien1985 , if you need further examples of tf 0.12 code , I'm happy to kick more your way.

**Comment by lukehoban** [comment]
Initial support for HCL2 (TF0.12) was added in https://github.com/pulumi/tf2pulumi/pull/160, and is now available as part of `tf2pulumi` `v0.8.0`.

We will be continuing to work on fleshing out more support for TF0.12, including follow-up work tracked in https://github.com/pulumi/tf2pulumi/issues/172.  Please do report issues if you see any limitations in the TF0.12 support for your codebase.
